crates/ide_assists/src/handlers/add_missing_match_arms.rs

```diff
@@ -86,7 +85,10 @@ pub(crate) fn add_missing_match_arms(acc: &mut Assists, ctx: &AssistContext) ->
         let missing_pats = variants
             .into_iter()
             .filter_map(|variant| {
-                Some((build_pat(ctx.db(), module, variant)?, variant.has_doc_hidden(ctx.db())))
+                Some((
+                    build_pat(ctx.db(), module, variant)?,
+                    variant.should_be_hidden(ctx.db(), module.krate()),
+                ))
```

Comment:
Fetching the crate of the variants looks a bit wrong to me, it would make more sense in my eyes to fetch enum's crate once, compare that and then skip the should_be_hidden checks.
So something like
```rs
let different_krate = module.krate() != enum_def.krate();
```

```suggestion
                Some((
                    build_pat(ctx.db(), module, variant)?,
                    different_crate && variant.should_be_hidden(ctx.db()),
                ))
```

Reply:
I agree that this would be better, but this isn't as easily extended to the tuple case, as the enum_def isn't available where I would need it to be

Reply:
Ah ye okay I see, then that's fine.